app/src/main/java/com/group80/uoftinder/GenerateCompatibilityList.java

```diff
@@ -0,0 +1,113 @@
+     */
+    public void generateCompatibilityList() {
+        compatibilityList = new ArrayList<>();
+        Map<String, Integer> compScores = calculateCompatibilityScores(allUsers);
+        for (int i = 0 ; i < users.size() ; i++) {
+            String maxKey = getMaxKey(compScores);
+            compatibilityList.add(maxKey);
+            compScores.remove(maxKey);
+        }
+//        curUser.setCompatibilityList(compatibilityList);
+//        recPresenterInterface.showUsers(compatibilityList);
+    }
+
+    /**
+     * Return the key that corresponds to the max value in userScore
+     * @param userScore: a map that maps the user ID to their compatibility score with the current
+     *                 user
+     * @return the key that corresponds to the max value in userScore
+     */
+    private String getMaxKey(Map<String, Integer> userScore) {
+        // TODO: change this to get key of max value
+        int maxValue = 0;
+        String maxKey = "";
+        for (String key : userScore.keySet()) {
+            Integer currentScore = userScore.get(key);
+            if (currentScore != null && currentScore > maxValue) {
+//                if (currentScore > maxValue) {
+                maxValue = currentScore;
+                maxKey = key;
+//                }
+            }
+
+        }
+        return maxKey;
+    }
+
+    /**
+     * Return a map of user ID to the user's compatibility score with the current user
+     * @param users: a list of all User in the database
+     * @return a map of user ID to the user's compatibility score with the current user
+     */
+    private Map<String, Integer> calculateCompatibilityScores(List<User> users) {
+        Map<String, Integer> compScores = new Hashtable<>();
+        for (User user : users) {
+            calculateCompatibiltyScore(compScores, user);
+        }
+        return compScores;
+    }
+
+    /**
+     * Mutate compScores to contain a map of compUser's user ID to their compatibility score with
+     * curUser
+     * @param compScores: a map of user IDs to their compatibility score with curUser
+     * @param compUser: the user to compare with curUser
+     */
+    private void calculateCompatibiltyScore(Map<String, Integer> compScores, User compUser) {
+        int userScore = compUser.getScore();
+        int compScore = usf.compare(curUserScore, userScore);
+        compScores.put(compUser.getUid(), compScore);
+    }
+
+    /**
+     * Removes the most compatible user (the first element) from compatibilityList and send it to
+     * the recPresenterInterface to display it
+     */
+    public void update() {
+        if (compatibilityList.size() > 0) {
+            String nextUserId = compatibilityList.get(0);
+            User nextUser = UserRealtimeDbFacade.getUser(nextUserId);
+            this.compatibilityList.remove(0);
+            curUser.setCompatibilityList(compatibilityList);
+            recPresenterInterface.displayUser(nextUser);
```

Comment:
Slightly confused by the logic here and would love more clarification. It seems that you have already removed the User that is about to be displayed on the screen. The only concern I have with this is that if the current user quits and app completely and rejoins before they make a decision about the displayed user, there's a chance that the compatibility list will no longer contain this displayed user when the app reloads? It seems more logical to me to do just include the last 3 lines in this if block that you have?

Reply:
I see the problem here, but how would including just the last 3 lines resolve this? Maybe I'm not completely understanding your thought process here, Helen.

From what I understand, maybe it would make more sense to get the next User from the compatibilityList _without_ removing them from the list, until the User moves on to the next person. This way, when the app is closed and reopened, we can retrieve the last displayed person from the 0th index of the compatibilityList.

Reply:
You raise a valid point, I will consider this and make the necessary changes.

Reply:
Raghav, I was basically thinking what you articulated. Nicholas and I called offline, and we thought it would be best to separate removing and displaying the first user in the list into two separate methods for the View/Presenter to call.

Reply:
Yep, this makes a lot of sense to me.